src/tzdb.rs

```diff
@@ -329,7 +375,117 @@ fn get_timezone_offset(db: &DataBlock, idx: usize) -> TimeZoneTransitionInfo {
 fn get_local_record(db: &DataBlock, idx: usize) -> LocalTimeTypeRecord {
     // NOTE: Transition type can be empty. If no transition_type exists,
     // then use 0 as the default index of local_time_type_records.
-    db.local_time_type_records[db.transition_types.get(idx).copied().unwrap_or(0)]
+    let idx = db.transition_types.get(idx).copied().unwrap_or(0);
+
+    let get = db.local_time_type_records.get(idx);
+    debug_assert!(get.is_some(), "tzif internal invariant violated");
+    get.copied().unwrap_or_default()
+}
+
+#[inline]
+fn get_transition_info(db: &DataBlock, idx: usize) -> TzifTransitionInfo {
+    let next = get_local_record(db, idx);
+    let transition_time = db.transition_times.get(idx);
+    debug_assert!(
+        transition_time.is_some(),
+        "tzif internal invariant violated"
+    );
+    let transition_time = transition_time.copied().unwrap_or_default();
+    let prev = if idx == 0 {
+        db.local_time_type_records[0]
+    } else {
+        get_local_record(db, idx - 1)
+    };
+    TzifTransitionInfo {
+        prev,
+        next,
+        transition_time,
+    }
+}
+
+/// Information obtained from the tzif file about a transition.
+#[derive(Debug)]
+struct TzifTransitionInfo {
+    /// The time record from before this transition
+    prev: LocalTimeTypeRecord,
+    /// The time record corresponding to this transition and dates after it
+    next: LocalTimeTypeRecord,
+    /// The UTC epoch seconds for the transition
+    transition_time: Seconds,
+}
+
+impl TzifTransitionInfo {
+    /// Get the previous offset as a number of seconds from
+    /// 1970-01-01 in local time as reckoned by the previous offset
+    fn transition_time_prev_epoch(&self) -> Seconds {
+        // You always add the UTC offset to get the local time;
+        // so a local time in PST (-08:00) will be `utc - 8h`
+        self.transition_time + self.prev.utoff
+    }
+    /// Get the previous offset as a number of seconds from
+    /// 1970-01-01 in local time as reckoned by the next offset
+    fn transition_time_next_epoch(&self) -> Seconds {
+        // You always add the UTC offset to get the local time;
+        // so a local time in PST (-08:00) will be `utc - 8h`
+        self.transition_time + self.next.utoff
+    }
+
+    /// Gets the range of local times where this transition is active
+    ///
+    /// Note that this will always be start..end, NOT prev..next: if the next
+    /// offset is before prev (e.g. for a TransitionKind::Overlap) year,
+    /// it will be next..prev.
+    ///
+    /// You should use .kind() to understand how to interpret this
+    fn offset_range_local(&self) -> Range<Seconds> {
```

Comment:
I'm not sure if we're producing the right type of range here. I think having it be `[start, end)` is correct, but I'm not sure.